src/utilities.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utilities used by the charm."""
+
+import logging
+import subprocess  # nosec B404
+import time
+from typing import Callable, Optional, Sequence, Type, TypeVar
+
+logger = logging.getLogger(__name__)
+
+# Return type of decorated function
+R = TypeVar("R")
+
+
+def retry(
+    exception: Type[Exception] = Exception,
+    tries: int = 1,
+    delay: float = 0,
+    max_delay: Optional[float] = None,
+    backoff: float = 1,
+    logger: logging.Logger = logger,
+) -> Callable[[Callable[..., R]], Callable[..., R]]:
+    """Parameterize the decorator for adding retry to functions.
+
+    Args:
+        exception: Exception type to be retried. Defaults to Exception.
```

Comment:
Is it worth re-documenting the default value? It is already noted in the function signature?